crates/bifrost/src/providers/replicated_loglet/remote_sequencer.rs

```diff
@@ -140,36 +134,43 @@ where
             payloads,
         };
 
-        let rpc_token = loop {
+        let loglet_commit = loop {
+            let permits = self
+                .record_permits
+                .clone()
+                .acquire_many_owned(len)
+                .await
+                .unwrap();
             match connection
-                .send(&self.sequencers_rpc.append, self.params.sequencer, msg)
+                .send(
+                    permits,
+                    &self.sequencers_rpc.append,
+                    self.params.sequencer,
+                    msg,
+                )
                 .await
             {
-                Ok(token) => break token,
+                Ok(loglet_commit) => break loglet_commit,
                 Err(err) => {
                     match err.source {
-                        NetworkError::ConnectError(_)
-                        | NetworkError::ConnectionClosed(_)
-                        | NetworkError::Timeout(_) => {
-                            // we retry to re-connect one time
+                        err @ NetworkError::Full => return Err(err.into()),
```

Comment:
We should never receive this error as you are using a async-blocking send on the channel.